src/bun.js/webcore/Response.zig

```diff
@@ -498,8 +530,77 @@ pub fn constructRedirect(
     try headers_ref.put(.Location, url_string_slice.slice(), globalThis);
     const ptr = bun.new(Response, response);
 
-    return ptr.toJS(globalThis);
+    const vm = globalThis.bunVM();
```

Comment:
This doesn't go here. We do not need to run this code on every single call to `Response.redirect`. This code should only be run when the Response is the BakeResponse.